wallet-new/server/Cardano/Wallet/API/V1/Swagger.hs

```diff
@@ -79,8 +79,16 @@ newDescr :: Typeable a => proxy a -> T.Text
 newDescr (p :: proxy a) =
     "A type represending an request for creating a(n) " <> renderType p <> "."
 
+-- | Derives all fields.
+allFieldsFromJSON :: forall a proxy. (Arbitrary a, ToJSON a) => proxy a -> Set T.Text
+allFieldsFromJSON _ =
```

Comment:
How come did you end up adding this new function? What was the problem in writing:

``` haskell
instance ToDocs AddressInfo where
  readOnlyFields   = readOnlyFieldsFromJSON
 descriptionFor _ = "An Address."
```

Reply:
I couldn't reasonably define `Update AddressInfo`, because it's fully read-only.
So couldn't use `readOnlyFieldsFromJSON`

Reply:
@adinapoli-iohk I can declare `type Update AddressInfo = ()`, it fits to current usages of `Update` type family in neat way.
But I don't know which other usages for `Update` do you have in mind, so can't say that such declaration would be correct in all usage cases. 
What do you think?

Reply:
Hey @Martoon-00 -- that's a valid suggestion. The `Update` (and `New`) type families were created for just 2 reasons:

- As an indirection to allow us to create better, more descriptive APIs in servant so that, for example, you can express at the type level you are creating a new `Address` by typing `New Address` rather than the `NewAddress` ad-hoc type. This also allows us to change the underlying type name without breaking the API;

- In the case of `Update` is also used to computed the `diff` of the two JSON object (the input, LHS one and the RHS one) to show in Swagger which fields are readonly (Swagger will add a nice `readOnly: true` description in the specs. What you propose sound sensible, I cannot think on top of my head to any problems it might create.

Your patch has something which concerns me more, though, hehe. I tried to inspect the generated swagger specs and for some reason the `spendingPassword` is marked as required whereas in our types is marked as a `Maybe` (quite rightly):

![screen shot 2017-12-01 at 13 05 12](https://user-images.githubusercontent.com/29383371/33487154-9e596f40-d6ac-11e7-9370-d29ac46bd5b4.png)

Do you know why is that? Am I staring at the wrong API?

Reply:
@adinapoli-iohk Note that the same applies to spec for `NewWallet`, while this PR didn't touch it.
Perhaps this behaviour derives from `sketchSchema`:
https://github.com/GetShopTV/swagger2/pull/48/files#diff-aa239390df4e45623cfb02ef393bdab4R309
AFAIS it just takes all fields it sees and marks them as required

Reply:
> Note that the same thing you mention applies to spec for NewWallet, while this PR didn't touch it.

Yea, you're right!

> AFAIS it just takes all fields it sees in JSON and marks them as required

Meh, that's a bit lame, but I guess this is what it is 😁